clients/src/main/java/org/apache/kafka/common/security/auth/PlaintextAuthenticationContext.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.security.auth;
+
+public class PlaintextAuthenticationContext implements AuthenticationContext {
+    public static final PlaintextAuthenticationContext INSTANCE = new PlaintextAuthenticationContext();
+
```

Comment:
We can currently set ACLs based on remote client IP, and it is possible to configure Principal based on remote IP with the old PrincipalBuilder (e.g. to set quotas). This doesn't seem to allow that.